From 0a22a9d9ba926651c0ad16bf080fb71d91e6b42f Mon Sep 17 00:00:00 2001
From: Brian Liu <brian.liu@freescale.com>
Date: Wed, 27 Oct 2010 15:16:41 +0800
Subject: [PATCH 089/498] ENGR00133039 CTS failure:
 android.hardware.cts.CameraTest#testAccessParameters

Need add some parameters in CameraHAL which are mandatory.

Signed-off-by: Brian Liu <brian.liu@freescale.com>
---
 libcamera/CameraHal.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/libcamera/CameraHal.cpp b/libcamera/CameraHal.cpp
index 0080d41..70b69c0 100755
--- a/libcamera/CameraHal.cpp
+++ b/libcamera/CameraHal.cpp
@@ -179,6 +179,14 @@ void CameraHal::initDefaultParameters()
     p.set(CameraParameters::KEY_SUPPORTED_FOCUS_MODES, tmpBuffer);
     p.set(CameraParameters::KEY_FOCUS_MODE, CameraParameters::FOCUS_MODE_AUTO);
 
+    p.set(CameraParameters::KEY_FOCAL_LENGTH, "4.31");
+    p.set(CameraParameters::KEY_HORIZONTAL_VIEW_ANGLE, "54.8");
+    p.set(CameraParameters::KEY_VERTICAL_VIEW_ANGLE, "42.5");
+    p.set(CameraParameters::KEY_EXPOSURE_COMPENSATION, "0");
+    p.set(CameraParameters::KEY_MAX_EXPOSURE_COMPENSATION, "0");
+    p.set(CameraParameters::KEY_MIN_EXPOSURE_COMPENSATION, "0");
+    p.set(CameraParameters::KEY_EXPOSURE_COMPENSATION_STEP, "0");
+
     memset(tmpBuffer, '\0', sizeof(*tmpBuffer));
     strncat((char*) tmpBuffer, (const char*) CameraParameters::ANTIBANDING_50HZ, PARAM_BUFFER);
     strncat((char*) tmpBuffer, (const char*) PARAMS_DELIMITER, PARAM_BUFFER);
-- 
1.8.0

